src/components/manage-registry/ManageRegistry.tsx

```diff
@@ -0,0 +1,228 @@
+  const [selectedIdentity, setSelectedIdentity] = useState<{
+    walletAddr: EvmAddress;
+    identityAddr: EvmAddress;
+  }>();
+  const [selectedToken, setSelectedToken] = useState<TokenNameItem>();
+  const { registry, registryIdentities } =
+    useTokenIdentityRegistry(selectedToken);
+  const [updateTxResult, setUpdateTxResult] = useState<string>();
+  const [updateTxError, setUpdateTxError] = useState<string>();
+  const { deployedTokens } = useContext(Eip3643Context);
+  const { accountEvm } = useWalletInterface();
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  useEffect(() => {
+    (deployedTokens as any).map((item: any) => {
+      const tokenAddress = item["args"]?.[0];
+      tokenAddress &&
+        readTokenOwner({}, tokenAddress).then((resOwner) => {
+          resOwner[0].toString().toLowerCase() === accountEvm?.toLowerCase() &&
+            readTokenName({}, tokenAddress).then((resName) => {
+              setOwnTokens((prev) => {
+                return [
+                  ...prev.filter((itemSub) => itemSub.address !== tokenAddress),
+                  {
+                    address: tokenAddress,
+                    name: resName[0],
+                  },
+                ];
+              });
+            });
+        });
+    });
+  }, [accountEvm, deployedTokens, setOwnTokens]);
+
+  const onEditIdentityRegistry = (identityRegistry: {
+    walletAddr: EvmAddress;
+    identityAddr: EvmAddress;
+  }) => {
+    onOpen();
+    setSelectedIdentity(identityRegistry);
+  };
+
+  const handleTokenSelect = (tokenAddr: string) => {
+    setSelectedToken(
+      ownTokens.find((token) => token.address === (tokenAddr as string)),
+    );
+  };
+
+  const selectedRegistryIdentity = registryIdentities.find(
+    (identity) => identity.walletAddr === selectedIdentity?.walletAddr,
+  );
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={onClose} size="xl">
+        <ModalOverlay />
+        <ModalContent width="80%" pt="8">
+          <ModalHeader mb="8">
+            <Text fontSize={22} fontWeight="800" align="center" mb="2">
+              Manage identity agents {"\b"}
+            </Text>
+            <Text fontSize={14} align="center">
+              Identity address: {selectedRegistryIdentity?.identityAddr}
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            {isAgents ? (
+              <ManageAgents
+                onClose={onClose}
+                setUpdateTxError={setUpdateTxError}
+                setUpdateTxResult={setUpdateTxResult}
+                registry={registry}
+              />
+            ) : (
+              <ManageIdentities
+                onClose={onClose}
+                isOpen={isOpen}
+                setUpdateTxError={setUpdateTxError}
+                setUpdateTxResult={setUpdateTxResult}
+                registry={registry}
+                selectedIdentity={selectedIdentity}
+              />
+            )}
```

Comment:
I have concerns with this way of mixing responsibilities of the component and coupling.
Please consider having 2 distinct components - `ManageAgentsContainer` and `ManageIdentitiesContainer` (names are just for the example), which would be separate and showing respective modals accordingly.
This would be easier to support, test and add new tabs in the future.
It would be more visible if they share any functionality, which can be extracted and reused further.

I'd also suggest showing error/success messages in the popups itself, without passing the setters inside the modal body components. But it would require user to close popup manually, as we close it automatically atm.

Reply:
Those two components: `<ManageAgents />` and `<ManageIdentities />` was done in the way they actually calling inside <ManageRegistry /> common component, i don't think we need any changes here. Also about this popup it's common thing we provide to the user opportunity to close it manually.